src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -44,8 +44,42 @@ def iot_hub_create(client, name, resource_group_name,
     return result
 
 
+def iot_hub_get_connection_string(client, name, resource_group_name, policy_name='iothubowner'):
+    access_policies = client.list_keys(resource_group_name, name).next()
+    if access_policies is None:
+        raise CLIError('No policy found from IoT Hub: {}.'.format(name))
+    logger.info('Shared Access Polices: %s', str(access_policies))
+    try:
+        access_policy = next(x for x in access_policies
+                             if policy_name.lower() == x.key_name.lower())
+    except StopIteration:
+        raise CLIError('No policy found with name {}'.format(policy_name))
+
+    connection_string = 'HostName={}.azure-devices.net;SharedAccessKeyName={};SharedAccessKey={}'.format(name,
+                                                                                                         policy_name,
+                                                                                                         access_policy.primary_key)
+    return {'connectionString': connection_string}
+
+
 def iot_device_create(client, resource_group_name, hub, device_id):
```

Comment:
>hub [](start = 51, length = 3)

Instead of using hub here and name in your connection string methods, if you consistently use the same parameter (i.e. hub_name) then you'll be able to more efficiently alias your parameters. Otherwise you'll have multiple entries doing the same thing because you have to alias both 'hub' and 'name' (and since name can refer to different things in different contexts, it can get confusing).